core/watcher/watchertest/watcher.go

```diff
@@ -82,6 +101,34 @@ func (w *WatcherC[T]) AssertChange() {
 	}
 }
 
+// AssertNChanges fails if it does not receive n changes before a long time has passed.
+func (w WatcherC[T]) AssertNChanges(n int) {
+	if n <= 1 {
+		w.c.Fatalf("n must be greater than 1")
+	}
+	received := 0
+	for {
+		select {
+		case _, ok := <-w.Watcher.Changes():
+			w.c.Check(ok, jc.IsTrue)
```

Comment:
This should be an assert, as you want it to fail if the channel is closed. Otherwise you'll interpret the received value as correct, which it's not.